doc-content/enterprise-only/openshift/offline-repo-proc.adoc

```diff
@@ -1,16 +1,54 @@
+endif::[]
+ifeval::["{context}"!="openshift-immutable"]
+. Prepare a Maven release repository to which you can write. The repository must allow read access without authentication. Your OpenShift environment must have access to this repository. You can deploy a Nexus repository manager in the OpenShift environment. For instructions about setting up Nexus on OpenShift, see https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/developer_guide/tutorials#nexus-setting-up-nexus[Setting up Nexus]. Use this repository as a separate mirror repository.
++
+Alternatively, if you use a custom external repository (for example, Nexus) for your services, you can use the same repository as a mirror repository.
+endif::[]
++
+. On the computer that has an outgoing connection to the public Internet, complete the following steps:
+.. Download the http://release-engineering.github.io/offliner/[Offliner tool].
+.. Download the `{PRODUCT_FILE}-offliner.txt` product deliverable file from the {PRODUCT_DOWNLOAD_LINK}[Software Downloads] page of the Red Hat Customer Portal.
+.. Enter the following command to download the required artifacts using the Offliner tool:
++
+[subs="attributes,verbatim,macros"]
+----
+java -jar offliner-1.3.jar -r https://maven.repository.redhat.com/ga/ -d /home/user/temp rhpam-7.3.1-offliner.txt
+----
++
+Replace `/home/user/temp` with an empty temporary directory on the computer.
+.. Upload all artifacts from the temporary directory to the Maven mirror repository that you prepared. You can use the https://github.com/simpligility/maven-repository-tools/tree/master/maven-repository-provisioner[Maven Repository Provisioner] utility to upload the artifacts.
+. If you developed any services outside {CENTRAL} and they have additional dependencies, upload the dependencies to the mirror repository. Optionally, on the computer that has an outgoing connection to the public Internet, you can use the following steps to prepare these dependencies automatically:
 .. Clear the local Maven cache directory (`~/.m2/repository`).
-.. Build the source of your services using the `mvn clean install` command.
-.. Upload all artifacts from the local Maven cache directory (`~/.m2/repository`) to the Maven repository that you prepared. You can use the https://github.com/simpligility/maven-repository-tools/tree/master/maven-repository-provisioner[Maven Repository Provisioner] utility to upload the artifacts.
+.. Build the source of your projects using the `mvn clean install` command.
+.. For every project, enter the following command to ensure that Maven downloads all runtime dependencies for all the artifacts generated by the project:
++
+[subs="attributes,verbatim,macros"]
+----
+mvn -e -DskipTests dependency:go-offline -f /path/to/project/pom.xml --batch-mode -Djava.net.preferIPv4Stack=true
+----
++
+Replace `/path/to/project/pom.xml` with the correct path to the `pom.xml` file of the project.
```

Comment:
@mramendi You are mentioning `services` but then you switch to projects. Could we please unify it across the document? 
Basically I am missing some sentence that would say that services can be maven projects or that they are maven projects in our case. Wdyt?

Reply:
Service can be a simple jar too.

Reply:
All right - done now, thanks! While any JAR must have a source, not all source is Maven so you do have a great point.

Reply:
Thank you. I am currently verifying the scenario. I will keep you posted.